url/url_test.cpp

```diff
@@ -634,6 +637,128 @@ int main() {
         etest::expect_eq(*url, url::Url{.scheme = "a", .host = url::Host{.type = url::HostType::Opaque}});
     });
 
+    etest::test("URL parsing: file url with base", [] {
+        url::UrlParser p;
+
+        std::optional<url::Url> file_base = p.parse("file:///usr/bin/vim");
+
+        etest::require(file_base.has_value());
+
+        etest::expect(p.parse("file:usr/bin/emacs", file_base).has_value());
```

Comment:
Can you expect_eq these against what this should actually parse to? I could delete the entirety of the file parsing and these 2 tests would still pass. This is both to have working examples and to make sure I don't break things if I touch the url parsing in the future. :P